Sources/Commands/Workspace.swift

```diff
@@ -494,35 +494,31 @@ public class Workspace {
         }
 
         // Load the state.
-        //
-        // FIXME: Build out improved file reading support.
-        try fopen(statePath) { handle in
-            let json = try JSON(bytes: ByteString(encodingAsUTF8: try handle.readFileContents()))
+        let json = try JSON(bytes: try localFileSystem.readFileContents(statePath))
```

Comment:
Can we cleanup CheckoutManager to be consistent with this too? Really they should share common basic serialization support, but it didn't seem worth trying to design that right now.